pimd/pim_sock.c

```diff
@@ -253,16 +311,65 @@ int pim_socket_mcast(int protocol, struct in_addr ifaddr, struct interface *ifp,
 	}
 
 	/* Set Tx socket DSCP byte */
-	if (setsockopt_ipv4_tos(fd, IPTOS_PREC_INTERNETCONTROL)) {
-		zlog_warn("can't set sockopt IP_TOS to PIM/IGMP socket %d: %s",
+	if (setsockopt_ipv6_tclass(fd, IPTOS_PREC_INTERNETCONTROL)) {
+		zlog_warn("can't set sockopt IP_TOS to PIMv6 socket %d: %s",
 			  fd, safe_strerror(errno));
 	}
 
+	return 0;
+}
+
+int pim_socket_mcast(int protocol, struct in_addr ifaddr, struct interface *ifp,
```

Comment:
It should be 
int pim_socket_mcast(int protocol, pim_addr ifaddr, struct interface *ifp,
		     uint8_t loop);

Reply:
Ack